torch/nn/functional.py

```diff
@@ -4922,6 +4922,7 @@ def multi_head_attention_forward(
     key_padding_mask: Optional[Tensor] = None,
     need_weights: bool = True,
     attn_mask: Optional[Tensor] = None,
+    _causal_mask: bool = False
```

Comment:
Just mentioning that doesn't really make the arguments private. It will still show up in the signature of the function on the doc page.
This function is not actually documented on the website so the point is moot-ish but you can see where it is mentioned in https://docs-preview.pytorch.org/90508/jit_builtin_functions.html#supported-pytorch-functions

Reply:
TBH, I don't think it should be private -- mirroring the concerns here from dan3, I think it will be preferable to give users the explicit option.